platform/openide.filesystems/src/org/openide/filesystems/FileUtil.java

```diff
@@ -2324,24 +2320,17 @@ private static boolean isArchiveFileImpl(final URL url, final boolean strict) {
     }
 
     private static Iterable<? extends ArchiveRootProvider> getArchiveRootProviders() {
-        if (archiveRootProviderCache == null) {
-            Lookup.Result<ArchiveRootProvider> res = archiveRootProviders.get();
-            if (res == null) {
-                res = new ProxyLookup(
-                    Lookups.singleton(new JarArchiveRootProvider()),
-                    Lookup.getDefault()).lookupResult(ArchiveRootProvider.class);
-                if (!archiveRootProviders.compareAndSet(null, res)) {
-                    res = archiveRootProviders.get();
-                    res.addLookupListener((ev) -> {
-                        archiveRootProviderCache = null;
-                    });
-                }
+        Lookup.Result<ArchiveRootProvider> res = archiveRootProviders.get();
+        if (res == null) {
+            res = new ProxyLookup(
+                Lookups.singleton(new JarArchiveRootProvider()),
+                Lookup.getDefault()).lookupResult(ArchiveRootProvider.class);
+            if (!archiveRootProviders.compareAndSet(null, res)) {
+                res = archiveRootProviders.get();
             }
-            archiveRootProviderCache = new LinkedList<>(res.allInstances());
         }
-        return archiveRootProviderCache;
+        return res.allInstances();
```

Comment:
This introduces a regression compared to https://github.com/apache/netbeans/commit/290420cfff1af3f72dd5eb86f864edd74ff78fe8

Reply:
I think I properly reverted it.. Thanks for catching this..